setup/lib/utils.js

```diff
@@ -0,0 +1,74 @@
+const { setFailed } = require('@actions/core');
+const os = require('os');
+const path = require('path');
+const execSync = require('child_process').execSync;
+const semver = require('semver');
+const core = require('@actions/core');
+
+// arch in [arm, x32, x64...] (https://nodejs.org/api/os.html#os_os_arch)
+// return value in [amd64, 386, arm]
+function mapArch(arch) {
+    const mappings = {
+        x32: '386',
+        x64: 'amd64'
+    };
+    return mappings[arch] || arch;
+}
+
+// os in [darwin, linux, win32...] (https://nodejs.org/api/os.html#os_os_platform)
+// return value in [darwin, linux, windows]
+function mapOS(os) {
+    const mappings = {
+        darwin: 'macOS',
+        win32: 'windows'
+    };
+    return mappings[os] || os;
+}
+
+function getDownloadObject(version) {
+    const platform = os.platform();
+    const filename = `notation_${version}_${mapOS(platform)}_${mapArch(os.arch())}`;
+    const extension = platform === 'win32' ? 'zip' : 'tar.gz';
+    const binPath = platform === 'win32' ? 'bin' : path.join(filename, 'bin');
+    const url = `https://github.com/notaryproject/notation/releases/download/v${version}/${filename}.${extension}`;
+    return {
+        url,
+        binPath,
+        filename
+    };
+}
+
+function getDownloadPluginObject(name, version) {
+    if (name == "notation-azure-kv") {
+        const platform = os.platform();
+        const filename = `${name}_${version}_${mapOS(platform)}_${mapArch(os.arch())}`;
+        const extension = platform === 'win32' ? 'zip' : 'tar.gz';
+        const url = `https://github.com/Azure/notation-azure-kv/releases/download/v${version}/${filename}.${extension}`;
+        const HOME = process.env.HOME;
+        const pluginPath = HOME + "/.config/notation/plugins/azure-kv"
+        return {
+            url,
+            filename,
+            pluginPath
+        };
+    } else { // Add logic for additional Notation plugins here
+        setFailed(`Plugin ${name} is not currently supported`)
+    }
+}
+
+function addPluginCert(keyName,keyId){
+    if (keyId.includes('vault.azure.net')){
+        const output = execSync(`notation key add --plugin azure-kv --id ${keyId} --default ${keyName}`, { encoding: 'utf-8' });
+        console.log('notation cert output:\n', output);
+    } // Add logic for additional Notation plugins here
+}
+
+function versionCheck(version){
+    const supportedVersion = '0.12.0-beta.1'
```

Comment:
Let's update the supported versions: https://github.com/notaryproject/notation/releases

Reply:
Further down in the thread @shizhMSFT mentioned replacing this with a checksum. What do you think about removing this and adding a checksum in a sperate PR later?

Reply:
> Further down in the thread @shizhMSFT mentioned replacing this with a checksum. What do you think about removing this and adding a checksum in a sperate PR later?

Yeah, we should check the SHA256 checksum after the download and before extracting. We could have a separate PR for it. And I think we should keep the version check? When the user input is just a Notation CLI version, then we should check the version before trying to download it to save the roundtrip.

Reply:
We need to maintain a `version`-to-`checksum` map. If a version is not in that map, it is "unsupported" and thus a checksum should be provided by the user. We can do that in another PR.

Reply:
> We need to maintain a `version`-to-`checksum` map. If a version is not in that map, it is "unsupported" and thus a checksum should be provided by the user. We can do that in another PR.

Oh yeah, good idea!

Reply:
I removed the `versionCheck` code. :)